sup/core.py

```diff
@@ -87,6 +95,18 @@ def log_updates(update, did_anything_change):
 
 sup_tasks = []
 
+def parse_config_file():
```

Comment:
I may contribute some edits of my own here since I have ideas about how `xconf` "should" work, but as long as it's working it's fine for now.

I'm not sure it's necessary to re-read the config every page-load, though actual complete reloads happen relatively infrequently so it's not like it's a load on the system. I'm just thinking the way to get everything to take effect after editing a config file should be to restart the service itself, for consistency.

Reply:
I removed the config re-read on page-load and left it only on service initialization. Happy (and curious) for any edits needed to standardize the use of `xconf`